packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -42,4 +50,63 @@ class ExpoImageModule(val context: ReactApplicationContext) : ReactContextBaseJa
       }
     }
   }
+
+  @ReactMethod
+  fun queryCache(urls: ReadableArray, promise: Promise) {
+    moduleCoroutineScope.launch {
+      try {
+        val resultMap = WritableNativeMap()
+        urls.toArrayList()
+            .filterIsInstance<String>()
+            .map { url -> async { Triple(url, isInCache(url, ImageCacheType.DISK), isInCache(url, ImageCacheType.MEMORY))  } }
+            .awaitAll()
+            .filter {
+              if (it.second == null || it.third == null) {
+                throw InterruptedException()
+              } else {
+                it.second == true || it.third == true
+              }
+            }
+            .forEach{
+              resultMap.putString(
+                  it.first,
+                  if (it.second == true && it.third == true) {
+                    "disk/memory"
+                  } else if (it.second == true) {
+                    "disk"
+                  } else {
+                    "memory"
+                  }
+              )
+            }
+        promise.resolve(resultMap)
+      } catch (e: Exception) {
+        promise.reject("ERR_IMAGE_PREFETCH_FAILURE", "Failed to read the cache: ${e.message}", e)
+      }
+    }
+  }
+
+  private suspend fun isInCache(url: String, cacheOption: ImageCacheType) : Boolean? {
```

Comment:
```suggestion
  private suspend fun isInCache(url: String, cacheOption: ImageCacheType) : Boolean {
```
(see comment about exceptions)